polaris/competition/_competition.py

```diff
@@ -0,0 +1,99 @@
+from datetime import datetime
+import os
+from typing import Optional, Union
+
+from pydantic import (
+    field_serializer,
+    field_validator,
+    ValidationInfo
+)
+from polaris.benchmark import BenchmarkSpecification
+from polaris.hub.settings import PolarisHubSettings
+from polaris.utils.types import AccessType, HubOwner, PredictionsType, TimeoutTypes, ZarrConflictResolution
+from polaris.utils.errors import InvalidCompetitionError
+
+
+class CompetitionSpecification(BenchmarkSpecification):
```

Comment:
This is the new `CompetitionBenchmark` class, to accommodate the differences between regular benchmarks and competition benchmarks, notably:
- Competition benchmarks have a start and end time, since they only accept results for the duration of the competition.
- We validate that all test rows have at least one value across the target columns to ensure there are no rows in the test set with missing test data.
- Evaluation is handled in a separate, standalone service. This is to ensure that test labels are never processed in the client so there is no way users can access them. Users will only see the results reported from the evaluation service, but the test labels never exist on their machines.
- Uploading to hub is handled in a separate, standalone service, because some pre-processing needs to be done on the dataset file before it is available (the test labels are separated and masked in the dataset, so they are not available until a competition is over.